src/resolve-names.cc

```diff
@@ -412,7 +412,21 @@ Result NameResolver::OnCatchExpr(TryExpr*, Catch* catch_) {
 }
 
 Result NameResolver::OnDelegateExpr(TryExpr* expr) {
-  ResolveLabelVar(&expr->delegate_target);
+  Var* delegate_target = &expr->delegate_target;
+  bool was_name = delegate_target->is_name();
+  ResolveLabelVar(delegate_target);
+  if (was_name) {
+    // Invalid label (of the immediate try-delegate block) used.
+    if (delegate_target->index() == 0) {
+      PrintError(&delegate_target->loc, "invalid label variable for try-delegate");
+      return Result::Ok;
+    }
+    // Adjust label by one, because a delegate instruction cannot target its
+    // immediately enclosing try block.
+    delegate_target->set_index(delegate_target->index() - 1);
+  }
+  // Pop the label here as a try-delegate has no `end` instruction.
```

Comment:
Note: I added this line in the latest revision, as this was another bug in this resolution code. (it is sufficiently tested by the `dump` test case I added---without the pop the `$try1` label was off because it came after another `try-delegate`.)